hasher-matcher-actioner/hmalib/common/content_models.py

```diff
@@ -0,0 +1,222 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import datetime
+from enum import Enum
+import typing as t
+
+from dataclasses import dataclass, field, asdict
+from mypy_boto3_dynamodb.service_resource import Table
+from boto3.dynamodb.conditions import Attr, Key, Or
+from botocore.exceptions import ClientError
+
+from hmalib.lambdas.api.middleware import JSONifiable
+from hmalib.models import DynamoDBItem
+
+
+@dataclass
+class ContentObjectBase(DynamoDBItem):
+    """
+    Base content obeject for ActionRecords and ContentObjects
+    Both of which used content_id as there PK
+    (This is similar to Hash and Match Records
+    but can easily be moved to it's own db table as they do not
+    use the other indexes)
+    """
+
+    content_id: str
+
+
+@dataclass
+class ActionEventRecord(ContentObjectBase, JSONifiable):
+    """
+    Record of action events in the action performer
+    keyed based on action_label and time of event.
+    For the time being the action_rules in the action message
+    being preformed are also stored as a list serialized json blobs
+    """
+
+    ACTION_TIME_PREFIX = "action_time#"
+    DEFAULT_PROJ_EXP = "PK, SK, ActionLabel, ActionRules"
+
+    time_performed: datetime.datetime
```

Comment:
nit: To be consistent with other 'time' related attributes, `performed_at` might be better suited.